modules/internal/ChapelTuple.chpl

```diff
@@ -216,6 +216,19 @@ module ChapelTuple {
     return __primitive("get svec member", this, i);
   }
 
+  pragma "no doc"
+  pragma "reference to const when const this"
+  pragma "star tuple accessor"
+  proc _tuple.this(i : bool) ref {
+    if !isHomogeneousTuple(this) then
+      compilerError("invalid access of non-homogeneous tuple by runtime value");
+    if boundsChecking then
+      if i < 0 || i > size-1 then
+        halt("tuple access out of bounds: ", i);
```

Comment:
Similarly here. If we are sure that this is our error, maybe we should make this halt clearer